uwblueprint/rowan-house
# Setup manage users admin page

## Implementation Description
This PR will setup the manage users page

[modified]
- make users and course separate routes (so that when we navigate back and forward, it includes these pages separately in the history)
- delete the `AdminDashboard` combined component and move to separate components in `ManageUsers` and `ManageCourses` pages
- change sidebar to  have tabs be composed of buttons

[new]
- Create `UserCard` component to render a spotlighted user's information. Add equivalent `NoUserCard` if no spotlighted user
- Create `SearchUsersBar` component for searching for user by email -- EXACT MATCH ONLY
- Create rest of users page to match Figma designs here -- https://www.figma.com/file/Yb3MvhfQAhfCjLg8cnNCoo/User-Management?node-id=112%3A61
  - Uses query for all users (and sets up relevant API things). Caches on user side which produces an edge case if page is not refreshed for significant time while user goes through sign up flow, reasonable mitigation by forcing query refresh within ~5 mins?

## Screenshots

https://user-images.githubusercontent.com/40974372/162696816-410c916f-f5a3-422f-a74b-f72879e1561f.mov

## What should reviewers focus on?
- Better to use another component instead of <Menu>?

*Outstanding work/questions*
- Order of fields is wrong, but are we ok with that
- should prob have a redirect if they go to `/admin`
- Make admin pages such as this one only accessible to admin users

https://user-images.githubusercontent.com/40974372/162700457-07e36dc3-0274-49b9-9f15-7b49948877d3.mov

- The search bar has some really weird behaviour -> on correct answer typed, loses focus on the input field. But this doesn't happen when a correct email is copied exactly

https://user-images.githubusercontent.com/40974372/162699022-72e7c6a1-56c3-4d9f-ae2b-bafb2cb3d68f.mov

- because the list loses focus when the input field is clicked, the result disappears and you have to delete/type it for it to reappear


## Testing Procedure
- What test(s) should we run to make sure your code works?
   - What ways could the input be wrong?
   - Is it secure? Could nefarious users access/break it? *be careful since all users information loaded in frontend, but not PII*


## Things to Note
- Additional dependencies/libraries you've added? => None
- Things you'd want to know when coming back to the code after a few months
  - *Check the users query edge case (see top section of summary of implementation)

## Checklist
